pages/api/docs/userSelectedRepos.ts

```diff
@@ -0,0 +1,83 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import rudderStackEvents from "../events";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+import { publishMessage } from "../../../utils/pubsub/pubsubClient";
+import { SetupReposArgs, removePreviousSelections, saveSelectedReposInDb } from "../../../utils/db/setupRepos";
+
+const UserSelectedRepos = async (req: NextApiRequest, res: NextApiResponse) => {
+	console.info("[UserSelectedRepos] Saving setup info in db...");
+	const jsonBody = req.body;
+    const session = await getServerSession(req, res, authOptions);
+	if (!session?.user.id) {
+		const eventProperties = { response_status: 401 };
+        rudderStackEvents.track("absent", "", 'user-selected-repos', { type: 'HTTP-401', eventStatusFlag: 0, eventProperties });
+		return res.status(401).json({ error: 'Unauthenticated' });
+	}
+    const userId = session.user.id;
+    if (!Array.isArray(jsonBody.info) || !jsonBody.installationId) {
+		console.error("[UserSelectedRepos] Invalid request body", jsonBody);
+		res.status(400).json({ "error": "Invalid request body" });
+		const eventProperties = { response_status: 400 };
+		rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-400', eventStatusFlag: 0, eventProperties });
+		return;
+	}
+    if (!userId) {
+        const eventProperties = { response_status: 500 };
+        rudderStackEvents.track(userId, "", 'user-selected-repos', { type: 'HTTP-500', eventStatusFlag: 0, eventProperties });    
+        res.status(500).json({error: "No user id in session"});
+        return;
+    }
+	const event_properties = {
+		repo_provider: jsonBody.info.length > 0 ? jsonBody.info[0].provider : "",
```

Comment:
Add a comment here that the flow of our setup process mandates that all the repositories selected by the user will be for the same provider.

Reply:
Since you are using the provider information at multiple places, it is better to create a `const` with this value (and your comment can be on that line itself) and use that `const` everywhere else. This will improve the readability.

Reply:
done